public/components/common/seach/autocomplete.tsx

```diff
@@ -0,0 +1,270 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+import './search.scss';
+import React, { createElement, Fragment, useEffect, useRef, useState } from 'react';
+import { render } from 'react-dom';
+import { h } from 'preact';
+
+import { autocomplete } from '@algolia/autocomplete-js';
+import { RAW_QUERY } from '../../../common/constants/explorer';
+import { IQueryBarProps } from './search';
+import { handlePplRequest } from '../../../requests/ppl';
+
+// Possible suggestions (hardcoded)
+const firstCommand = [{ label: 'search' }, { label: 'source' }];
+
+const pipeCommands = [
+  { label: 'dedup' },
+  { label: 'eval' },
+  { label: 'fields' },
+  { label: 'head' },
+  { label: 'rare' },
+  { label: 'rename' },
+  { label: 'sort' },
+  { label: 'stats' },
+  { label: 'top' },
+  { label: 'where' },
+];
+
+const statsCommands = [
+  { label: 'count()' },
+  { label: 'sum()' },
+  { label: 'avg()' },
+  { label: 'max()' },
+  { label: 'min()' },
+];
+
+const indices = [
+  { label: 'opensearch_dashboards_sample_data_ecommerce' },
+  { label: 'opensearch_dashboards_sample_data_flights' },
+  { label: 'opensearch_dashboards_sample_data_logs' },
+];
+
+let inFieldsCommaLoop = false;
+let hasFields = false;
+const fieldsFromBackend = [];
+
+// Function to grab suggestions
+export function getSuggestions(str: string, http) {
+  const splittedModel = str.split(' ');
```

Comment:
just a question on the high level logic, PPL is separated by pipes, why split on space instead of `|`? splitting by | should give you a list of commands which can then be parsed on words or chars

Reply:
I split on space to list suggestions for each specific word. Since there's only a few commands, I have them all in an array and whenever the user types out a word, the array of commands is filtered by the most recent word. I also wanted to be able to check previous words; for example, `search` must be followed by `source =`, etc. I've just been adding more checks to my giant if else block as I continue with the project, but the plan is to refactor the grammar check to something more robust and coherent (split by pipes, then for the first element in the array the user must type search or source and an index? use properties to keep track of selected index or commands? something like that). Definitely open to suggestions, I've been adding checks to my code as I go along.

Reply:
I see. even splitting by `|` has a possibility to break if user has something like `source=index field="a|b"`. I think many parsers/compilers reads by character into tokens, but i'm not sure if full parsing of the query is needed for autocomplete and i have zero experience in these before. just worried that the big `if-else` implementation could get too complicated to maintain, just a thought